python/cells.cpp

```diff
@@ -428,179 +502,170 @@ void register_cells(pybind11::module& m) {
-            "rev_pot"_a=pybind11::none(),
-            "rev_pot_method"_a=pybind11::none(),
+            pybind11::arg_v("ion", "name of the ion species."),
+            pybind11::arg_v("int_con", pybind11::none(), "initial internal concentration [mM]."),
+            pybind11::arg_v("ext_con", pybind11::none(), "initial external concentration [mM]."),
+            pybind11::arg_v("rev_pot", pybind11::none(), "reversal potential [mV]."),
+            pybind11::arg_v("method",  pybind11::none(), "method for calculating reversal potential."),
             "Set the global default propoerties of ion species named 'ion'.\n"
             "Species concentrations and reversal potential can be overridden on\n"
             "specific regions using the paint interface, while the method for calculating\n"
             "reversal potential is global for all compartments in the cell, and can't be\n"
-            "overriden locally.\n"
-            " ion:     name of ion species.\n"
-            " int_con: initial internal concentration [mM].\n"
-            " ext_con: initial external concentration [mM].\n"
-            " rev_pot: initial reversal potential [mV].\n"
-            " rev_pot_method:  method for calculating reversal potential.")
+            "overriden locally.")
         .def_readwrite("catalogue", &global_props_shim::cat, "The mechanism catalogue.")
         .def("__str__", [](const global_props_shim& p){return to_string(p);});
 
-    // arb::cable_cell
 
-    pybind11::class_<arb::cable_cell> cable_cell(m, "cable_cell",
-        "Represents morphologically-detailed cell models, with morphology represented as a\n"
-        "tree of one-dimensional cable segments.");
-    cable_cell
-        .def(pybind11::init(
-            [](const arb::morphology& m, const label_dict_proxy& labels) {
-                return arb::cable_cell(m, labels.dict);
-            }), "morphology"_a, "labels"_a)
-        .def(pybind11::init(
-            [](const arb::segment_tree& t, const label_dict_proxy& labels) {
-                return arb::cable_cell(arb::morphology(t), labels.dict);
-            }),
-            "segment_tree"_a, "labels"_a,
-            "Construct with a morphology derived from a segment tree.")
-        .def_property_readonly("num_branches",
-            [](const arb::cable_cell& c) {return c.morphology().num_branches();},
-            "The number of unbranched cable sections in the morphology.")
-        // Set cell-wide properties
-        .def("set_properties",
-            [](arb::cable_cell& c,
+    pybind11::class_<arb::decor> decor(m, "decor",
+            "Description of the decorations to be applied to a cable cell, that is the painted,\n"
+            "placed and defaulted properties, mecahanisms, ion species etc.");
+    decor
+        .def(pybind11::init<>())
+        .def(pybind11::init<const arb::decor&>())
+        // Set cell-wide default values for properties
+        .def("set_property",
```

Comment:
At first thought, it seems better to have the same name across the C++ and the Python interfaces, i.e. `set_property` or `set_default`. Is there a reason to keep them distinct?

Reply:
The set_property and set_ion calls were part of the Python API for global properties and cable_cell before, and this here just reflects this.

For Python, interfaces like the following
`set_ion(ion_name, {named properties})`
`set_property([named properties])`
 were easier to define and implement than a `set_default` interface for setting both cable properties and  ion species values with named arguments. 

We could have a "typey" interface:
`set_default(arbor.ion(ion_name, [named properties])`
But such an interface for membrane potential, capacitance, etc would be overly verbose and awkward (my opinion), and a solution like the following
```Python
c.set_default(arbor.ion('ca', 'reversal_potential'=12))
c.set_default(Cm=0.1, Vm=-20)
```
is hard to implement.